### PR TITLE
Removed pgen workspaces. Workspaces is now Generator. Generator has t…

### PR DESCRIPTION
**Pull request by sarahmarshy**
…he ability to produce iterator object of generated projects. Tests were updated to reflect these changes.

### CODE REVIEW

**Comment by 0xc0170** [comment]
I am going to review the code. I noticed Travis still fails, it's because next() for iterators was changed, so we should use the one which is compatible in both versions (next(x)), more here http://docs.pythonsprints.com/python3_porting/py-porting.html#iterators

File: project_generator/generate.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+                for name, records in self.projects_dict['projects'].items():
+                    if type(records) is dict:
+                        # workspace
+                        for n, r in records.items():
+                            project = Project(n, load_yaml_records(uniqify(flatten(r))), self)
+                            yield project
+                            projects.append(projects)
+                        self.workspaces[name] = ProjectWorkspace(name, projects, self)
+                    else:
+                        yield Project(name, load_yaml_records(uniqify(flatten(records))), self)
```
Comment (0xc0170) [inline project_generator/generate.py]: Why it yields Project, but not workspace?

Comment (sarahmarshy) [inline project_generator/generate.py]: Workspace has no clean command

Comment (sarahmarshy) [inline project_generator/generate.py]: Or build

**Pull request merged**